cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatusTool.java

```diff
@@ -0,0 +1,379 @@
+    System.out.println("Run this tool on the Master cluster:");
+    System.out.println("#ReplicationStatusTool -m <master cluster id> -o <output filepath>");
+    System.out.println("Then run on the Slave Cluster:");
+    System.out.println("#ReplicationStatusTool -m <master cluster id> -i <file copied from Master>\n");
+    return;
+  }
+
+  private static void processMasterCluster() throws IOException {
+    //Get checksum from Snapshot files
+    masterChecksum = getClusterChecksum();
+    masterShutdownTime = getShutdownTime();
+
+    if (masterShutdownTime == 0L) {
+      System.out.println("CDAP Shutdown time not available. Please run after CDAP has been shut down.");
+      return;
+    }
+
+    //Scan the table and write to output file
+    writeMasterStatusFile(getMapFromTable(ReplicationConstants.ReplicationStatusTool.WRITE_TIME), outputStatusFileName);
+    System.out.println("Copy the file " + outputStatusFileName + " to the Slave Cluster and run tool there.");
+  }
+
+  private static void processSlaveCluster() throws Exception {
+    slaveClusterID = localClusterID;
+    Map<String, Long> masterTimeMap = readMasterStatusFile(inputStatusFileName);
+
+    if (masterShutdownTime == 0L) {
+      System.out.println("Could not read CDAP Shutdown Time from input file."
+                           + " Please make sure CDAP has been shutdown on the Master and rerun the tool on Master.");
+      return;
+    }
+    if (masterChecksum == null) {
+      System.out.println("Could not read Master Checksum from input file.");
+    }
+    if (masterTimeMap == null || masterTimeMap.size() == 0) {
+      LOG.info("No region information found in the input file");
+    }
+
+    checkHBaseReplicationComplete(masterTimeMap);
+    //checkSnapshotReplicationComplete(masterChecksum);
+  }
+
+  private static void checkHBaseReplicationComplete(Map<String, Long> masterTimeMap) throws Exception {
+    // Get replicate Time from table for all regions
+    Map<String, Long> slaveTimeMap = getMapFromTable(ReplicationConstants.ReplicationStatusTool.REPLICATE_TIME);
+
+    //Check all regions are accounted on both clusters
+    if (masterTimeMap.size() != slaveTimeMap.size()) {
+      System.out.println("Number of regions on the Master and Slave Clusters do not match."
+                         + " HBase Replication is NOT complete.");
+      return;
+    }
+
+    //Check the maps for all regions
+    for (Map.Entry<String, Long> writeTimeEntry: masterTimeMap.entrySet()) {
+      String region = writeTimeEntry.getKey();
+      long writeTime = writeTimeEntry.getValue();
+      long replicateTime = slaveTimeMap.get(region);
+
+      if (!isReplicationComplete(masterShutdownTime, writeTime, replicateTime)) {
+        System.out.println("HBase Replication is NOT complete. Region:"
+                             + region + " behind by "
+                             + (writeTime - replicateTime) + "ms.");
+        return;
```

Comment:
rather than returning here, it would be better to loop through so that everything that is behind is printed out